nilearn/maskers/multi_nifti_masker.py

```diff
@@ -542,12 +543,75 @@ def transform(self, imgs, confounds=None, sample_mask=None):
 
         """
         check_is_fitted(self)
-        if not hasattr(imgs, "__iter__") or isinstance(imgs, str):
-            return self.transform_single_imgs(imgs)
```

Comment:
for all multimaskers confounds and sample_mask were not applied if a nifti-like object was passed

when a list of nifti-like object was passed, they could not handle properly the expected diversity on inputs array, dataframe, csv file...